src/delta.rs

```diff
@@ -70,16 +73,20 @@ where
             }
         } else if line.starts_with("diff --git ") {
             painter.paint_buffered_lines();
-            state = State::FileMeta;
+            state = State::GitFileMeta;
+            painter.set_syntax(parse::get_file_extension_from_diff_line(&line));
+        } else if line.starts_with("diff -u ") || line.starts_with("diff -U") {
```

Comment:
Shall we include the trailing space after both?

Reply:
The `-U` parameter takes an argument (the number of context lines)
So something like `diff -U3 ...` is a valid command

I avoided a regex matcher because `-U` is the only parameter that begins with a capital U, so it seemed unnecessary overhead.

Let me know what you think. I'm happy to either extend the check and look for spaces or digits, or make a regex pattern

Reply:
> The `-U` parameter takes an argument

Oh, I see. Makes sense then, I like your simple and efficient `starts_with` implementation.

But, sorry if I'm being stupid, I don't see a string like `diff -u` in unified diff output, do you?

```
$ diff -u <(echo a) <(echo b)
--- /dev/fd/63  2019-11-20 10:08:51.973225512 -0500
+++ /dev/fd/62  2019-11-20 10:08:51.973585959 -0500
@@ -1 +1 @@
-a
+b
```

```
$ diff --version
diff (GNU diffutils) 3.6
```

However, the output pasted by @fdcds in the intro of #53 does show a string like that, so I'm a bit confused. Is there a different version of `diff` that outputs such a string? (On a Mac the system diff and the one from the coreutils brew package seem to be the same.) Seems the same on linux:

```
$ docker run ubuntu bash -c 'diff -u <(echo a) <(echo b); uname -a; diff --version'
--- /dev/fd/63  2019-11-20 15:17:14.345904000 +0000
+++ /dev/fd/62  2019-11-20 15:17:14.345904000 +0000
@@ -1 +1 @@
-a
+b
Linux 7c3c98a876c6 4.9.184-linuxkit #1 SMP Tue Jul 2 22:58:16 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
diff (GNU diffutils) 3.6
```

Reply:
Yeah.. This is what I commented on https://github.com/dandavison/delta/issues/53, it wasn't very clear what was happening and I couldn't replicate it. I was very confused at first. But 
I did manage to find out what he meant, though. You need to `diff -u <directory1> <directory2>`, this will output `diff -u ...` before each file, or `Only in <directory1>` for files that do not have a name match

Reply:
Right, I see, thanks for explaining.